src/library/recording/dlgrecording.cpp

```diff
@@ -120,7 +113,11 @@ void DlgRecording::setFocus() {
 }
 
 void DlgRecording::refreshBrowseModel() {
-    m_browseModel.setPath(mixxx::FileAccess(mixxx::FileInfo(m_recordingDir)));
+    saveCurrentViewState();
+    QString recordingDir = m_pRecordingManager->getRecordingDir();
+    // Hack: append "/" so the model stores the complete path the model state key
+    m_browseModel.setPath(mixxx::FileAccess(mixxx::FileInfo(
```

Comment:
`BrowseTabelModel::setPath()` should be responsible for normalizing the path as needed. Otherwise this only works for DlgRecording, but not for anyone else.

Reply:
`BrowseTabelModel::setPath()` gets `/path/to/some/dir/` from the BrowseFeature and this works for both the model key as well as passing the path to BrwoseThread. Besides, "anyone else" is fictional since no other feature uses the BrowseTableModel.
The actual "issue" is that we store `/path/to/recording/directory` in ConfigKey(RECORDING_PREF_KEY, "Directory") without trailing separator (and I didn't want to touch how RecordingManager handles this)
So, IMO this 'hack' for the exceptional DlgRecording is okay IMO, compared to complicating the code elsewhere (and I thought about how to to do this _safely_ and with little effort  more than twice).

Reply:
Qt normally doe not use a trailing separator for directories. I have not yet understand why this makes a difference here. Can you add another sentence that explains what happens otherwise? 
How about assert that case inside setPath?

Reply:
See the commit message of e79a977b8270d391e861e9b33c63d72a4c1dd957

The existing code
https://github.com/mixxxdj/mixxx/blob/f45552763b28061409588c229b8937c7c3bb348f/src/library/browse/browsetablemodel.cpp#L212-L219

* Model view state keys are stored without trailing /
* `/any/path/passed/from/BrowseModel/` has trailing / and becomes `/any/path/passed/from/BrowseModel`
which is fine because any accidentally passed `/path/to/MUSIC/file.ext` would properly be stripped to `/path/to/MUSIC`
* Without trailing / the rec path from the config`/home/me/Music/Recordings` would become `/home/me/Music`

Reply:
Comments about the current code should be added to the code instead of hiding this information in a commit message.

Reply:
Okay, I'll add those to browsetablemodel.cpp
Is the issue resolved for you then?

Reply:
_ping_

Reply:
FWIW the / is appended in Browsefeature, even though it is not needed to populate the track view (the Rec view was working fine without /)
https://github.com/mixxxdj/mixxx/blob/main/src/library/browse/browsefeature.cpp#L420-L425

Reply:
There is also the issue that BrowseTableModel::setPath(mixxx::FileAccess path) is either badly named or broken. 
Because it expects a "file" location. 

I think this will solve the issue by its roots: 

```
void BrowseTableModel::setPath(mixxx::FileAccess path) { 
     if (path.info().hasLocation()) { 
         if (isFile()) {
             m_currentDirectory = path.info().locationPath();
         } else {
             m_currentDirectory = path.info().location();
         }
     } else { 
         m_currentDirectory = QString(); 
     } 
     m_pBrowseThread->executePopulation(std::move(path), this); 
 } 
 ```

Reply:
That is exactly what I wanted to avoid.

Back to square one:
Browse passes all paths with trailing / (manually appended, for no obvious reason).
Removing the / was actually just a cosmetic issue for the model key string and `path.info().locationPath()` a simple way to
* check if path "is valid" and
* remove /

Rec is the only feature that _was_ passing a fixed path _without_ trailing slash.
My change to append / there, too, is only working around `.locationPath()` chopping the last dir part.
We may simply check for `path.info().isDir()` and use the submitted `path` then.